pkg/cloud/cloud.go

```diff
@@ -154,8 +156,32 @@ func (c *cloud) GetMetadata() MetadataService {
 	return c.metadata
 }
 
-func (c *cloud) CreateAccessPoint(ctx context.Context, volumeName string, accessPointOpts *AccessPointOptions) (accessPoint *AccessPoint, err error) {
+func (c *cloud) CreateAccessPoint(ctx context.Context, volumeName string, accessPointOpts *AccessPointOptions, usePvcName bool) (accessPoint *AccessPoint, err error) {
 	efsTags := parseEfsTags(accessPointOpts.Tags)
+
+	//if usePvcName is true, check for AP with same Root Directory exists in efs
+	// if found reuse that AP
+	if usePvcName {
+		existingAP, err := c.findAccessPointByPath(ctx, volumeName, accessPointOpts)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to create access point: %v", err)
```

Comment:
Why are we already returning "failed to create" here? We are not creating AP until later in `res, err := c.efs.CreateAccessPointWithContext(ctx, createAPInput)`